vars/kanikoExecute.groovy

```diff
@@ -0,0 +1,127 @@
+     * You can find more details about the Docker credentials in the [Docker documentation](https://docs.docker.com/engine/reference/commandline/login/).
+     */
+    'dockerConfigJsonCredentialsId',
+    /** @see dockerExecute */
+    'dockerEnvVars',
+    /** @see dockerExecute */
+    'dockerOptions',
+    /** @see dockerExecute */
+    'dockerImage'
+]
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+/**
+ * Executes a [Kaniko](https://github.com/GoogleContainerTools/kaniko) build for creating a Docker container.
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+
+        final script = checkScript(this, parameters) ?: this
+
+        // load default & individual configuration
+        Map config = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .use()
+
+        new Utils().pushToSWA([
+            step: STEP_NAME
+        ], config)
+
+        def buildOptions = new GStringTemplateEngine().createTemplate(config.containerBuildOptions).make([config: config, env: env]).toString()
+
+        if (!buildOptions.contains('--destination')) {
+            if (config.containerImageNameAndTag) {
+                buildOptions += " --destination ${config.containerImageNameAndTag}"
+            } else {
+                buildOptions += " --no-push"
+            }
+        }
+
+        dockerExecute(
+            script: script,
+            containerCommand: config.containerCommand,
+            containerShell: config.containerShell,
+            dockerEnvVars: config.dockerEnvVars,
+            dockerImage: config.dockerImage,
+            dockerOptions: config.dockerOptions
+        ) {
+            // prepare kaniko container for running with proper Docker config.json and custom certificates
+            sh """#!${config.containerShell}
+${config.containerPreparationCommand}
+${getCertificateUpdate(config.customTlsCertificateLinks)}
+"""
+
+            def uuid = UUID.randomUUID().toString()
+            if (config.dockerConfigJsonCredentialsId) {
+                // write proper config.json with credentials
+                withCredentials([file(credentialsId: config.dockerConfigJsonCredentialsId, variable: 'dockerConfigJson')]) {
+                    writeFile file: "${uuid}-config.json", text: readFile(dockerConfigJson)
+                }
+            } else {
+                // empty config.json to allow anonymous authentication
+                createJsonCall = "echo '{\"auths\":{}}' > /kaniko/.docker/config.json"
```

Comment:
Seems obsolete to me as we `mv` to this location later on and `createJsonCall` is never used.

Reply:
good catch :-)